### ISSUE TITLE
Microservices using App Configuration via providers in AKS 

### ISSUE THREAD

**Issue by vijayanand-msft** [Azure/AppConfiguration#629]
Based on MSDocs - https://docs.microsoft.com/en-us/azure/azure-app-configuration/integrate-kubernetes-deployment-helm it is possible to use Helm to integrate APP configuration in AKS. 
Is there any recommendation/ disadvantages in using App configuration with App settings and the secrets accessed directly within the Api's via the providers. 
Instead of managing the secrets and app settings file exported from app configuration and mounting it as environment variables, the app can directly access the secrets and app setting via the providers as mentioned in the docs(https://docs.microsoft.com/en-us/azure/azure-app-configuration/quickstart-aspnet-core-app?tabs=core6x).  I am planning to use this approach with the Microservices architecture in AKS. So with the proper cache expiration when a new api or the existing api is scaled out the configurations are loaded within the api.  
By doing this way even the AKS admin cannot view the secrets loaded from the app configuration .

In short, 

- Create app settings and secrets with keyvault reference in App configuration
- Use .net providers for accessing the settings in microservices with cache expiration
- Deploy the microservices in AKS
- Access the values from App configuration within the Microservices hosted in AKS

Do you see any issues with this approach?

**Comment by zhenlan** [comment]
@vijayanand-msft there are no issues with your plan. 

Your microservices get access to settings, key vault references, and feature flags by directly using the App Config provider in your microservices. Everything can also be updated dynamically without redeployment. App Config provider has built-in caching and refreshing capabilities.

The Helm template approach is helpful for users who cannot (or are not willing to) make code changes to their applications. App Config data are brought to AKS configmap via Helm upgrade. The applications continue accessing configuration as before, so it's transparent to them. Users run Helm upgrade when they want their application to pick up new configuration data.

**Comment by vijayanand-msft** [comment]
Thank you for clarifying @zhenlan :)

**Comment by zhenlan** [comment]
Good suggestion, @vijayanand-msft. I have a [PR ](https://github.com/MicrosoftDocs/azure-docs/pull/91556) out to update the documentation.

**Comment by vijayanand-msft** [comment]
@zhenlan  Thank you for doing this PR :)
